src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -289,6 +289,37 @@
  * }
  * </pre>
  * <p>
+ * To configure the check so that it limit IllegalType to validate field types only:
+ * </p>
+ * <pre>
+ * &lt;module name="IllegalType"&gt;
+ *   &lt;property name=&quot;illegalClassNames&quot; value=&quot;java.util.Optional&quot;/&gt;
+ *   &lt;property name=&quot;tokens&quot; value=&quot;VARIABLE_DEF&quot;/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="SuppressionXpathSingleFilter"&gt;
```

Comment:
Please see documentation for https://checkstyle.sourceforge.io/config_filters.html#SuppressionXpathSingleFilter.  We need use the `checks` property (and I would even use the `id` property in this example, too). As this is, it would  suppress all violations from `METHOD_DEF` (and child) tokens, from any check.  The check `id` could be something like `IllegalTypeOptionalAsField` or something.

Reply:
done

Reply:
Resolved.